molecule_libvirt/playbooks/create.yml

```diff
@@ -0,0 +1,259 @@
+    default_memory_size: "1"
+    default_vcpu: "1"
+    default_ssh_port: "22"
+    mac: "52:54:00:ab:cd:"  # first 5 digits
+    temp_folder: "/tmp"
+
+  tasks:
+
+    - name: Ensure default vm folder exists
+      file:
+        path: "{{ libvirt_default_image_path }}"
+        state: directory
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: Allow qemu process read qcow2 images in home directory
+      acl:
+        path: "{{ user_home_directory }}"
+        entity: qemu
+        etype: user
+        permissions: x
+        state: present
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: "Create network {{ molecule_network_name }}"
+      virt_net:
+        command: define
+        name: "{{ molecule_network_name }}"
+        xml: "{{ lookup('template', 'network/network.xml.j2') }}"
+        uri: "{{ qemu_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+      vars:
+        qemu_uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+
+    - name: "Start network {{ molecule_network_name }}"
+      virt_net:
+        command: start
+        name: "{{ molecule_network_name }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+
+    - name: Get qcow2 image
+      get_url:
+        url: "{{ image_url | default(default_image_url) }}"
+        dest: "{{ libvirt_default_image_path }}/{{ item.name }}.qcow2"
+        mode: '0660'
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Generate OpenSSH key pair
+      openssh_keypair:
+        path: "{{ molecule_private_key_path }}"
+        size: 2048
+      register: ssh_key
+      delegate_to: "localhost"
+
+    - name: Create temporary folder
+      file:
+        path: "{{ temp_folder }}/{{ item.name }}-molecule"
+        state: directory
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create user-data from template
+      template:
+        src: user-data.j2
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/user-data"
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
`with_items` has been changed for `loop`.